### DISCUSSION TITLE
Is there any library-internal timing/performance logging we can turn on?

### DISCUSSION THREAD

**Discussion by chrisbenincasa** [mikro-orm/mikro-orm#5736] [Q&A]
I'm trying to optimize some portions of an application I'm working on. I'm working with SQLite using the better-sqlite driver. I have a query that ends up selecting about 6.5k results (not huge) from one table with quite a few joins. It takes about 900ms on my Macbook. However, the performance timer I've wrapped around the `find` call iinto mikro-orm reports the operation took over 4s. This is backed up by the log timestamps too. I'm wondering if there are some further debugging mechanisms for library internals I can enable to start to track down where the slowdown is coming from (or perhaps you have an idea)

**Comment by B4nan** [comment]
> about 6.5k results (not huge)

Mapping 6,5k entities is actually quite huge. Maybe there are some easy wins on how to optimize that, complete repro welcome, it shouldn't take 4s unless your entities are really huge.

One thing you can try is setting `flushMode: 'commit'` in the ORM config (not just in find options, that wouldn't help). By doing that, the scalar properties wont be redefined as get/set pairs since we don't need to track their changes eagerly (to be able to support the default auto flush mode).

**Reply by chrisbenincasa** [reply]
gotcha, makes sense. I meant huge in terms of the result set - wasn't sure how that sized in terms of the library. Makes sense that it could take some time to map these back. I can try to get a simpler repro, it might take some time to whittle down the example so it's not just exactly what the app data model is. 

I'll try that out, thanks!

**Reply by chrisbenincasa** [reply]
Do you think trying to limit which columns we select would help here? Like if there's a large text column we don't need, could that speed things up? Or is it simply the mapping step itself and not necessarily the size of the underlying data. Or perhaps a little of both?

**Reply by B4nan** [reply]
I think it should help too (but probably not much), simply because your objects won't occupy that much memory. You will need to test things out, don't expect me to give you the right answer without seeing the code myself. If you at least share the entity definition, maybe I can think of something else, but often the perf issues are coming from relations and propagation of their values (and that is impossible to diagnose/optimize/fix without a complete repro).

**Reply by chrisbenincasa** [reply]
That is fair. I wasn't looking for a definitive answer, merely just wanted to discuss, considering you have a ton more context and knowledge about how the library works internally. 

I'll see if I can get a representative repro setup over the next few days.
